src/doblib/action.py

```diff
@@ -273,6 +277,37 @@ def _date(self, rec, name, **kw):
         upper = kw.get("upper", date.today())
         return lower + timedelta(days=random.randint(0, (upper - lower).days))
 
+    def _many2one(self, env, rec, name, **kw):
+        """Return a value for Many2one fields depending on the arguments
+
+        * Replacement of the reference with a random record from a search with
+          a `domain` filter
+        """
+        domain = kw.get("domain", [])
+        records = env[rec._name].search(domain)
+        if not records:
+            return False
+        return random.choice(records.ids)
+
+    def _many2many(self, env, rec, name, **kw):
+        """Return a value for Many2many fields depending on the arguments
+
+        * Replacement of the references with random records from a search with
+          a `domain` filter. If `num` is specified, return `num` random records.
+        """
+        domain = kw.get("domain", [])
+        records = env[rec._name].search(domain)
+
+        if not records:
+            return [(5,)]
+
+        num = kw.get("num", None)
+        if num is None:
+            return [(6, 0, [random.choice(records.ids)])]
```

Comment:
We know the len of the current value and can use that here and also use sample instead of choice. I would say: If num is None use len(records) otherwise use num records.  Can we name the option `length`?